In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [5]:
df = pd.read_csv('student-por.csv', sep=";")

In [6]:
df.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        4   0  11  11  
1      5        3      3     1     1      3        2   9  11  11  
2      4        3      2     2     3      3        6  12  13  12  
3      3        2      2     1     1      5        0  14  14  14  
4      4        3      2     1     2      5        0  11  13  13  

[5 rows x 33 columns]

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 649 entries, 0 to 648
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      649 non-null    object
 1   sex         649 non-null    object
 2   age         649 non-null    int64 
 3   address     649 non-null    object
 4   famsize     649 non-null    object
 5   Pstatus     649 non-null    object
 6   Medu        649 non-null    int64 
 7   Fedu        649 non-null    int64 
 8   Mjob        649 non-null    object
 9   Fjob        649 non-null    object
 10  reason      649 non-null    object
 11  guardian    649 non-null    object
 12  traveltime  649 non-null    int64 
 13  studytime   649 non-null    int64 
 14  failures    649 non-null    int64 
 15  schoolsup   649 non-null    object
 16  famsup      649 non-null    object
 17  paid        649 non-null    object
 18  activities  649 non-null    object
 19  nursery     649 non-null    object
 20  higher    

Let's break down the meaning of each column in your dataset:

*   **school**: Student's school (GP - Gabriel Pereira or MS - Mousinho da Silveira)
*   **sex**: Student's sex (F - female or M - male)
*   **age**: Student's age (from 15 to 22)
*   **address**: Student's home address type (U - urban or R - rural)
*   **famsize**: Family size (LE3 - less or equal to 3 or GT3 - greater than 3)
*   **Pstatus**: Parent's cohabitation status (T - living together or A - apart)
*   **Medu**: Mother's education (0 - none, 1 - primary education (4th grade), 2 - 5th to 9th grade, 3 - secondary education, 4 - higher education)
*   **Fedu**: Father's education (0 - none, 1 - primary education (4th grade), 2 - 5th to 9th grade, 3 - secondary education, 4 - higher education)
*   **Mjob**: Mother's job (at_home, health, other, services, teacher)
*   **Fjob**: Father's job (at_home, health, other, services, teacher)
*   **reason**: Reason to choose this school (home, reputation, course, other)
*   **guardian**: Student's guardian (mother, father, other)
*   **traveltime**: Home to school travel time (1 - <15 min, 2 - 15 to 30 min, 3 - 30 min to 1 hour, 4 - >1 hour)
*   **studytime**: Weekly study time (1 - <2 hours, 2 - 2 to 5 hours, 3 - 5 to 10 hours, 4 - >10 hours)
*   **failures**: Number of past class failures (n if 1<=n<3, else 4)
*   **schoolsup**: Extra educational support (yes or no)
*   **famsup**: Family educational support (yes or no)
*   **paid**: Extra paid classes within the course subject (yes or no)
*   **activities**: Extra-curricular activities (yes or no)
*   **nursery**: Attended nursery school (yes or no)
*   **higher**: Wants to take higher education (yes or no)
*   **internet**: Internet access at home (yes or no)
*   **romantic**: With a romantic relationship (yes or no)
*   **famrel**: Quality of family relationships (1 - very bad to 5 - excellent)
*   **freetime**: Free time after school (1 - very low to 5 - very high)
*   **goout**: Going out with friends (1 - very low to 5 - very high)
*   **Dalc**: Workday alcohol consumption (1 - very low to 5 - very high)
*   **Walc**: Weekend alcohol consumption (1 - very low to 5 - very high)
*   **health**: Current health status (1 - very bad to 5 - very good)
*   **absences**: Number of school absences (from 0 to 93)
*   **G1**: First period grade (numeric from 0 to 20)
*   **G2**: Second period grade (numeric from 0 to 20)
*   **G3**: Final grade (numeric from 0 to 20)

0   school      649 non-null    object

 1   sex         649 non-null    object

 2   age         649 non-null    int64

 3   address     649 non-null    object

 4   famsize     649
 non-null    object

 5   Pstatus     649
 non-null    object

 6   Medu        649
 non-null    int64

 7   Fedu        649
 non-null    int64

 8   Mjob        649
 non-null    object

 9   Fjob        649
 non-null    object

 10  reason      649
 non-null    object

 11  guardian    649
 non-null    object

 12  traveltime  649
 non-null    int64


 13  studytime   649
 non-null    int64

 14  failures    649
 non-null    int64

 15  schoolsup   649
 non-null    object

 16  famsup      649
 non-null    object

 17  paid        649
 non-null    object

 18  activities  649
 non-null    object

 19  nursery     649
 non-null    object

 20  higher      649
 non-null    object

 21  internet    649
 non-null    object

 22  romantic    649
 non-null    object

 23  famrel      649
 non-null    int64

 24  freetime    649
 non-null    int64

 25  goout       649
 non-null    int64

 26  Dalc        649
 non-null    int64

 27  Walc        649
 non-null    int64

 28  health      649
 non-null    int64

 29  absences    649
 non-null    int64

 30  G1          649
 non-null    int64

 31  G2          649 non-null    int64

 32  G3          649 non-null    int64

In [10]:
from scipy.stats import f_oneway
gp_g3 = df[df['school']=='GP']['G3']
ms_g3 = df[df['school']=='MS']['G3']

f_statistic, p_value = f_oneway(gp_g3, ms_g3)

print(f"F-statistic: {f_statistic}")
print(f"P-value: {p_value}")

if p_value < 0.05:
    print("There is a significant differenece in test scores between GP and MS schools.")
else:
    print("There is no significant difference in test scores between GP and MS schools.")

F-statistic: 56.890676863371326
P-value: 1.5661990923002604e-13
There is a significant differenece in test scores between GP and MS schools.
